src/connection/connection_v2.ts

```
@@ -108,20 +107,78 @@ export class ConnectionV2 extends BaseConnection {
     query: string,
     executeQueryOptions: ExecuteQueryOptions = {}
   ): Promise<StreamStatement> {
-    const { response } = await this.prepareAndExecuteQuery(query, {
-      ...executeQueryOptions,
-      settings: {
-        ...executeQueryOptions?.settings,
-        output_format: OutputFormat.JSON_LINES
-      }
-    });
+    const { response } = await this.prepareAndExecuteQuery(
+      query,
+      {
+        ...executeQueryOptions,
+        settings: {
+          ...executeQueryOptions?.settings,
+          output_format: OutputFormat.JSON_LINES
+        }
+      },
+      true
+    );
 
     return new StreamStatement({
       response,
       executeQueryOptions
     });
   }
 
+  async execute(
+    query: string,
+    executeQueryOptions: ExecuteQueryOptions = {}
+  ): Promise<Statement> {
+    if (this.options.preparedStatementParamStyle === "fb_numeric") {
+      return this.executePreparedStatement(query, executeQueryOptions);
+    }
+    return super.execute(query, executeQueryOptions);
+  }
+
+  private async executePreparedStatement(
+    query: string,
+    executeQueryOptions: ExecuteQueryOptions
+  ): Promise<Statement> {
+    const { text } = await this.executeQuery(
+      query,
+      this.getExecuteQueryOptionsForPreparedStatement(executeQueryOptions)
+    );
+
+    return new Statement(this.context, {
+      query: query,
+      text,
+      executeQueryOptions
+    });
+  }
+
+  private getExecuteQueryOptionsForPreparedStatement(
+    executeQueryOptions: ExecuteQueryOptions
+  ): ExecuteQueryOptions {
+    let queryParameters;
+    if (!executeQueryOptions.parameters) {
+      queryParameters = Object.entries(
+        executeQueryOptions.namedParameters || {}
+      ).map(([key, value]) => ({ name: key, value }));
+    } else if (!executeQueryOptions.namedParameters) {
+      queryParameters = executeQueryOptions.parameters.map((value, index) => ({
+        name: `$${index + 1}`,
+        value
+      }));
+    } else {
```

Should we throw an error if neither `parameters` nor `namedParameters` are provided?

---

I think this is just so we don't mix parameters and namedParameters for server-side prepared statement. It should work with no parameters set (i.e. a normal query) if a user is to reuse a server-side connection for a simple query.

---

I'll add a test for this.